src/afl-fuzz-cmplog.c

```diff
@@ -175,19 +178,32 @@ void init_cmplog_forkserver(afl_state_t *afl) {
 
   if (afl->frk_srv.exec_tmout) {
 
-    it.it_value.tv_sec = ((afl->frk_srv.exec_tmout * FORK_WAIT_MULT) / 1000);
-    it.it_value.tv_usec = ((afl->frk_srv.exec_tmout * FORK_WAIT_MULT) % 1000) * 1000;
+    fd_set readfds;
 
-  }
+    FD_ZERO(&readfds);
+    FD_SET(afl->cmplog_fsrv_st_fd, &readfds);
+    timeout.tv_sec = ((afl->frk_srv.exec_tmout * FORK_WAIT_MULT) / 1000);
+    timeout.tv_usec =
+        ((afl->frk_srv.exec_tmout * FORK_WAIT_MULT) % 1000) * 1000;
 
-  setitimer(ITIMER_REAL, &it, NULL);
+    int sret =
+        select(afl->cmplog_fsrv_st_fd + 1, &readfds, NULL, NULL, &timeout);
 
-  rlen = read(afl->cmplog_fsrv_st_fd, &status, 4);
+    if (sret == 0) {
 
-  it.it_value.tv_sec = 0;
-  it.it_value.tv_usec = 0;
+      if (afl->cmplog_forksrv_pid > 0) {
```

Comment:
This also always will be true.
Either remove the check or add an assert /FATAL to spot future bugs